R/epidemic_ebola.R

```diff
@@ -103,10 +110,13 @@ prob_discrete_erlang <- function(shape, rate) {
 #' pharmaceutical or non-pharmaceutical interventions applied to the infection's
 #' parameters, such as the transmission rate, over the epidemic.
 #' See [intervention()] for details on constructing rate interventions.
-#'
-#' Currently, only interventions on the transmission rate `beta` are supported,
-#' and should be passed as `list(beta = intervention(type = "rate", ...))`.
 #' Defaults to `NULL`, representing no interventions on model parameters.
+#'
+#' @param time_dependence A named list where each name
+#' is a model parameter (see `infection`), and each element is a function with
+#' the first two arguments being the current simulation `time`, and `x`, a value
+#' that is dependent on `time` (`x` represents a model parameter).
```

Comment:
I found this a bit confusing so I'll suggest the following rewording according to how I understand the code.
```suggestion
#' @param time_dependence A named list where each name
#' is a model parameter (see `infection`), and each element is a function with
#' the first two arguments fixed as `time`, and `x`, followed by other arguments to the function. `time` is used internally to refer to the model time at which the parameter value
#' will be changed to `x`.
```

Reply:
Should this allow for multiple breakpoints?

Reply:
Not sure what you mean by multiple breakpoints, but it is possible to pass a function that checks the value of time and changes the effect like so. Users can implement multiple cases if they wish, but if they are primarily seeking to model broad, step-wise changes to model parameters they would be better off with a `<rate_intervention>`. The time-dependence functionality works better for modelling small, continuous increments, such as seasonality.

```r
f = function(time, x) ifelse(time < 100, x, x^2)
```